src/internal/m365/service/onedrive/mock/handlers.go

```diff
@@ -359,29 +356,73 @@ func (en EnumerateDriveItemsDelta) EnumerateDriveItemsDelta(
 	return iterator.nextDelta()
 }
 
-type DriveDeltaEnumerator struct {
-	DriveID      string
+func (en EnumerateDriveItemsDelta) DrivePager() *apiMock.Pager[models.Driveable] {
+	drives := []models.Driveable{}
+
+	for _, dp := range en.DrivePagers {
+		drives = append(drives, dp.Drive.Able)
+	}
+
+	return &apiMock.Pager[models.Driveable]{
+		ToReturn: []apiMock.PagerResult[models.Driveable]{
+			{Values: drives},
+		},
+	}
+}
+
+func (en EnumerateDriveItemsDelta) Drives() []*DeltaDrive {
+	drives := []*DeltaDrive{}
+
+	for _, dp := range en.DrivePagers {
+		drives = append(drives, dp.Drive)
+	}
+
+	return drives
+}
+
+type DeltaDrive struct {
+	ID   string
+	Able models.Driveable
+}
+
+func Drive(driveSuffix ...any) *DeltaDrive {
+	driveID := id("drive", driveSuffix...)
+
+	able := models.NewDrive()
+	able.SetId(ptr.To(driveID))
+	able.SetName(ptr.To(name("drive", driveSuffix...)))
+
+	return &DeltaDrive{
+		ID:   driveID,
+		Able: able,
+	}
+}
+
+func (dd *DeltaDrive) NewEnumer() *DeltaDriveEnumerator {
+	copy := &DeltaDrive{}
+	*copy = *dd
```

Comment:
is a copy here strictly necessary? Also this looks like it may end up being a shallow copy since `dd.Able` is likely a pointer

Reply:
no, not strictly.  I just wanted to have the little extra safety. Avoidig accidental dereferencing helps me sleep at night.  You're right that `able` is still exposed here, but since we tend to use that with only a getter interface, I feel a little more calm about that one.